src/main/java/ladder/domain/ladder/LadderLines.java

```diff
@@ -72,19 +74,9 @@ public boolean equals(Object o) {
             return false;
         }
         LadderLines that = (LadderLines) o;
-        return isEqualLadderLines(that);
-    }
-
-    private boolean isEqualLadderLines(LadderLines that) {
-        long count = IntStream.range(0, this.ladderLines.size())
-                .filter(index -> isNotEqualLadderLine(that, index))
-                .count();
-
-        return count == 0;
-    }
+        AbstractCollection<LadderLine> ladderLines = (AbstractCollection<LadderLine>) this.ladderLines;
```

Comment:
`AbstractCollection` 으로 타입을 바꾸지 않아도, 적절한 `equals` 구현체를 사용할 것 같아요!
```suggestion
        return this.ladderLines.equals(that.ladderLines);
```